sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -110,10 +116,33 @@ class EventData(object):
 
     """
 
-    def __init__(self, body=None):
-        # type: (Union[str, bytes, List[AnyStr]]) -> None
+    @overload
+    def __init__(self, *, data: bytes, content_type: str, **kwargs):
+        ...
+
+    @overload
+    def __init__(self, body: Optional[Union[str, bytes, List[AnyStr]]] = None) -> None:
+        ...
+
+    def __init__(
+        self,
+        body: Optional[Union[str, bytes, List[AnyStr]]] = None,
+        *,
+        data: bytes = None,
+        content_type: str = None,
+        **kwargs
+    ) -> None:
         self._last_enqueued_event_properties = {}  # type: Dict[str, Any]
         self._sys_properties = None  # type: Optional[Dict[bytes, Any]]
+
+        if kwargs:
+            raise TypeError(f"__init__() got an unexpected keyword argument '{list(kwargs.keys())[0]}'")
+
+        if data:
+            if body:
```

Comment:
I think if we combine, like this:
```
if data and body:
  warnings.warn('overwrite')
  body = data
```
then it will not work if `body=None` and `data='x'`. body ends up as None. If we do this:
```
if data and body:
  warnings.warn(overwrite)
body = data
```
it will not work for `data=None` and `body='x'`.  body ends up as None.

Sorry, maybe I'm missing something obvious though? 😕

Reply:
could be the following?

```python
if data and body:
  warnings.warn(overwrite)
body = data or body
```